### PR TITLE
Support deployments/workloads probe overrides

### PR DESCRIPTION
**Pull request by skonto**
## Proposed Changes

* Customers are facing slow environments and need to adjust timeouts and other settings in probes.
* Some settings can be adjusted on the fly via changing the deployments eg. a deployment uses the probe defaults and given that the operator will not touch those (manifest uses apply) it does the trick. However this is a hack and we need a proper way to change probes and cover cases where reconciliation will touch the settings.

**Release Note**

```release-note
Support is added for readiness, liveness probe overrides.
```

### CODE REVIEW

**Comment by skonto** [comment]
@houshengbo @nak3

File: pkg/apis/operator/base/common.go
Comment (matzew) [inline pkg/apis/operator/base/common.go:289]: very valid to have these as overrides too! Good catch!

**Comment by matzew** [comment]
Let's have a knative/docs issue as well, so we describe this valuable feature!

**Comment by matzew** [comment]
LGTM

**Comment by skonto** [comment]
I will create a doc PR for this too once it is approved. @houshengbo gentle ping.

File: pkg/apis/operator/base/common.go
Comment (pierDipi) [inline pkg/apis/operator/base/common.go:339]: Should we prevent setting the [`ProbeHandler` field](https://github.com/skonto/operator/blob/23892c03b5a5caeb2e41691a4bff41248903dd73/vendor/k8s.io/api/core/v1/types.go#L2462) and allow only these fields https://github.com/skonto/operator/blob/23892c03b5a5caeb2e41691a4bff41248903dd73/vendor/k8s.io/api/core/v1/types.go#L2194-L2226.

File: pkg/apis/operator/base/common.go
Comment (matzew) [reply pkg/apis/operator/base/common.go:340]: > For v1, a more podspec-like structure (with a subset of fields) would be easier to work with.

yeah, currently is hard...

**Comment by houshengbo** [comment]
Do we really need to all fields under `livenessProbes`???
If it is just a few fileds, like TimeoutSeconds, InitialDelaySeconds, etc, we do not need to add all the other fields. Just pick the ones related to the `*** Seconds`, that are necessary.

File: pkg/apis/operator/base/common.go
Comment (skonto) [reply pkg/apis/operator/base/common.go:339]: I didnt want to limit people on how they can use this but we can.

File: pkg/apis/operator/base/common.go
Comment (skonto) [reply pkg/apis/operator/base/common.go:340]: Yes I would prefer a pod spec per workload to apply as an override.

**Comment by skonto** [comment]
> If it is just a few fileds, like TimeoutSeconds, InitialDelaySeconds, etc, we do not need to add all the other fields. Just pick the ones related to the *** Seconds, that are necessary.

Anything that is timeout related is useful as a starting point.

File: pkg/apis/operator/base/common.go
Comment (houshengbo) [reply pkg/apis/operator/base/common.go:339]: Let's not assume the options for users. If we plan to add some, what is the use case? Otherwise, limit to what is necessary. Drop ProbeHandler at least.

Comment (houshengbo) [reply pkg/apis/operator/base/common.go:339]: @skonto

**Comment by houshengbo** [comment]
Just keep 
```
InitialDelaySeconds int32 `json:"initialDelaySeconds,omitempty" protobuf:"varint,2,opt,name=initialDelaySeconds"`
	
	TimeoutSeconds int32 `json:"timeoutSeconds,omitempty" protobuf:"varint,3,opt,name=timeoutSeconds"`
	
	PeriodSeconds int32 `json:"periodSeconds,omitempty" protobuf:"varint,4,opt,name=periodSeconds"`
	
	SuccessThreshold int32 `json:"successThreshold,omitempty" protobuf:"varint,5,opt,name=successThreshold"`
	
	FailureThreshold int32 `json:"failureThreshold,omitempty" protobuf:"varint,6,opt,name=failureThreshold"`
	
	TerminationGracePeriodSeconds
```
in the CRDs. Remove the others. @skonto

**Comment by skonto** [comment]
@houshengbo kept those six fields. Adapted the PR.

**Comment by houshengbo** [comment]
@skonto Plz do another rebase. I will merge this pr.

**Comment by skonto** [comment]
@houshengbo sorry missed it I was off. Let's see if tests pass.

**Comment by skonto** [comment]
fixed crds /unhold

**Review by matzew — APPROVED** [review]
/lgtm
/approve

/hold
feel free to unhold

**Pull request merged**
